### PR TITLE
Misc compile fixes

### LINKED ISSUE #2698: Build errors on Linux
I built RPCS3 a while back, and it worked. Just tried to re-pull today, and...well...

```
[29/29] Building CXX object rpcs3/CMakeFiles/rpcs3.dir/rpcs3.cpp.o
FAILED: /home/ryan/stuff/Downloads/zapcc-20170226-190308-1.0/bin/zapcc++   -DCMAKE_BUILD -DGLX_GLXEXT_PROTOTYPES -DGL_GLEXT_PROTOTYPES -DLLVM_AVAILABLE -DWXUSINGDLL -D_FILE_OFFSET_BITS=64 -D__WXGTK__ -DwxGUI -fdiagnostics-color -std=c++14 -fexceptions -pthread -g -isystem /usr/lib/x86_64-linux-gnu/wx/include/gtk2-unicode-3.0 -isystem /usr/include/wx-3.0 -I../rpcs3/../3rdparty/libpng -I../rpcs3/../3rdparty/ffmpeg/linux/x86_64/include -I/usr/include/AL -I/usr/lib/llvm-4.0/include -I../rpcs3/../3rdparty/pugixml/src -I../rpcs3 -I../rpcs3/Loader -I../rpcs3/Crypto -I../rpcs3/.. -I../rpcs3/../Utilities/yaml-cpp/include -I../rpcs3/../asmjit/src/asmjit -I../rpcs3/../3rdparty/GSL/include -I../rpcs3/../rsx_program_decompiler/rsx_decompiler -I../rpcs3/../rsx_program_decompiler/shader_code -I../rpcs3/../Vulkan/Vulkan-LoaderAndValidationLayers/include -I../rpcs3/../Vulkan/glslang/glslang/Public -I../rpcs3/../3rdparty/GL -I../rpcs3/../3rdparty/stblib -I../rpcs3/../3rdparty/cereal/include    -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -ftemplate-depth=1024 -msse -msse2 -mcx16 -mssse3  -Qunused-arguments -include "/media/ryan/stuff/rpcs3/build/rpcs3/cotire/rpcs3_CXX_prefix.hxx" -MMD -MT rpcs3/CMakeFiles/rpcs3.dir/rpcs3.cpp.o -MF rpcs3/CMakeFiles/rpcs3.dir/rpcs3.cpp.o.d -o rpcs3/CMakeFiles/rpcs3.dir/rpcs3.cpp.o -c ../rpcs3/rpcs3.cpp
In file included from /media/ryan/stuff/rpcs3/build/../rpcs3/rpcs3.cpp:29:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLGSRender.h:3:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLHelpers.h:11:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/OpenGL.h:43:
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLProcTable.h:174:13: error: unknown type name 'PFNGLGETTEXTUREIMAGEPROC'; did you mean 'PFNGLGETTEXTUREIMAGEEXTPROC'?
OPENGL_PROC(PFNGLGETTEXTUREIMAGEPROC, GetTextureImage);
            ^~~~~~~~~~~~~~~~~~~~~~~~
            PFNGLGETTEXTUREIMAGEEXTPROC
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/OpenGL.h:42:27: note: expanded from macro 'OPENGL_PROC'
#define OPENGL_PROC(p, n) p n
                          ^
/usr/include/GL/glew.h:8370:28: note: 'PFNGLGETTEXTUREIMAGEEXTPROC' declared here
typedef void (GLAPIENTRY * PFNGLGETTEXTUREIMAGEEXTPROC) (GLuint texture, GLenum target, GLint level, GLenum format, GLenum type, GLvoid *pixels);
                           ^
In file included from /media/ryan/stuff/rpcs3/build/../rpcs3/rpcs3.cpp:29:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLGSRender.h:3:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLHelpers.h:11:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/OpenGL.h:43:
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLProcTable.h:185:13: error: unknown type name 'PFNGLTEXTUREBUFFERRANGEPROC'; did you mean 'PFNGLTEXTUREBUFFERRANGEEXTPROC'?
OPENGL_PROC(PFNGLTEXTUREBUFFERRANGEPROC, TextureBufferRange);
            ^~~~~~~~~~~~~~~~~~~~~~~~~~~
            PFNGLTEXTUREBUFFERRANGEEXTPROC
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/OpenGL.h:42:27: note: expanded from macro 'OPENGL_PROC'
#define OPENGL_PROC(p, n) p n
                          ^
/usr/include/GL/glew.h:6055:28: note: 'PFNGLTEXTUREBUFFERRANGEEXTPROC' declared here
typedef void (GLAPIENTRY * PFNGLTEXTUREBUFFERRANGEEXTPROC) (GLuint texture, GLenum target, GLenum internalformat, GLuint buffer, GLintptr offset, GLsizeiptr size);
                           ^
In file included from /media/ryan/stuff/rpcs3/build/../rpcs3/rpcs3.cpp:29:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLGSRender.h:3:
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLHelpers.h:1284:12: warning: 20 enumeration values not handled in switch: 'stencil', 'depth', 'red'... [-Wswitch]
                        switch (format_)
                                ^
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLHelpers.h:1527:15: error: use of undeclared identifier 'glTextureBufferRange'
                                __glcheck glTextureBufferRange(id(), gl_format_type, buf.id(), offset, length);
                                          ^
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLHelpers.h:1866:17: warning: enumeration value 'textureBuffer' not handled in switch [-Wswitch]
                                switch (rhs.get_target())
                                            ^
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLHelpers.h:2064:14: warning: enumeration value 'vertex' not handled in switch [-Wswitch]
                                        switch (shader_type)
                                                ^
In file included from /media/ryan/stuff/rpcs3/build/../rpcs3/rpcs3.cpp:29:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLGSRender.h:5:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:14:
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLRenderTargets.h:323:14: warning: comparison of unsigned expression >= 0 is always true [-Wtautological-compare]
                if (offset >= 0)
                    ~~~~~~ ^  ~
In file included from /media/ryan/stuff/rpcs3/build/../rpcs3/rpcs3.cpp:29:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLGSRender.h:5:
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:54:13: warning: enumeration values 'f64', 'ubyte_3_3_2', and 'ubyte_2_3_3_rev' not handled in switch [-Wswitch]
                                switch (type_)
                                        ^
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:285:6: error: use of undeclared identifier 'glGetTextureImage'; did you mean 'OpenGL::GetTextureImage'?
                                        glGetTextureImage(vram_texture, 0, (GLenum)format, (GLenum)type, pbo_size, nullptr);
                                        ^~~~~~~~~~~~~~~~~
                                        OpenGL::GetTextureImage
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLProcTable.h:174:39: note: 'OpenGL::GetTextureImage' declared here
OPENGL_PROC(PFNGLGETTEXTUREIMAGEPROC, GetTextureImage);
                                      ^
In file included from /media/ryan/stuff/rpcs3/build/../rpcs3/rpcs3.cpp:29:
In file included from /media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLGSRender.h:5:
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:285:6: error: invalid use of non-static data member 'GetTextureImage'
                                        glGetTextureImage(vram_texture, 0, (GLenum)format, (GLenum)type, pbo_size, nullptr);
                                        ^~~~~~~~~~~~~~~~~
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:764:15: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
                        if (texptr = m_rtts.get_texture_from_render_target_if_applicable(texaddr))
                            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:764:15: note: place parentheses around the assignment to silence this warning
                        if (texptr = m_rtts.get_texture_from_render_target_if_applicable(texaddr))
                                   ^
                            (                                                                    )
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:764:15: note: use '==' to turn this assignment into an equality comparison
                        if (texptr = m_rtts.get_texture_from_render_target_if_applicable(texaddr))
                                   ^
                                   ==
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:770:15: warning: using the result of an assignment as a condition without parentheses [-Wparentheses]
                        if (texptr = m_rtts.get_texture_from_depth_stencil_if_applicable(texaddr))
                            ~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:770:15: note: place parentheses around the assignment to silence this warning
                        if (texptr = m_rtts.get_texture_from_depth_stencil_if_applicable(texaddr))
                                   ^
                            (                                                                    )
/media/ryan/stuff/rpcs3/rpcs3/Emu/RSX/GL/GLTextureCache.h:770:15: note: use '==' to turn this assignment into an equality comparison
                        if (texptr = m_rtts.get_texture_from_depth_stencil_if_applicable(texaddr))
                                   ^
                                   ==
7 warnings and 5 errors generated.
ninja: build stopped: subcommand failed
```

Tested with both Clang 4.0 and Zapcc 5.0 (which is basically Clang, but different).

At a glance, it looks like something that can basically be fixed with some `#ifdef`'s and explicit namespaces (`OpenGL::`)...

### PR DESCRIPTION
**Pull request by kirbyfan64**
Fixes #2698. I think.

This basically consists of:

- Adding various missing includes.
- Adding a missing `using namespace` (required to use C++14 std::string literals).
- <strike>Compile fixes for OpenGL 4.4, which doesn't support `ARB_direct_state_access`. At all.</strike> Yup, that was stupid. Nevermind.

### CODE REVIEW

File: rpcs3/Crypto/utils.cpp [hunk 1]
```diff
@@ -2,8 +2,9 @@
 // Licensed under the terms of the GNU GPL, version 3
 // http://www.gnu.org/licenses/gpl-3.0.txt
 
-#include "utils.h"
-#include <stdio.h>
+#include "utils.h"
+#include <cstring>
+#include <stdio.h>
```
Comment (janisozaur) [inline rpcs3/Crypto/utils.cpp]: Why not `<cstdio>`?

**Comment by kd-11** [comment]
Opengl extensions dont work like that; youre working around header definitions rather than driver support. If you get compile errors because your linux header doesnt support 4.5 update it to the latest libglew. We cannot exclude functionality because of outdated headers.

**Comment by kirbyfan64** [comment]
@kd-11 Sorry, that makes sense. I removed that commit.

File: rpcs3/Crypto/utils.cpp (hunk 1 above)
Comment (kirbyfan64) [inline rpcs3/Crypto/utils.cpp]: Not sure. That line was already there (I'm not sure why that appeared as a changed line?).

**Pull request merged**